oxd-server/src/main/java/org/gluu/oxd/server/TracingUtil.java

```diff
@@ -0,0 +1,99 @@
+import zipkin2.reporter.okhttp3.OkHttpSender;
+
+public class TracingUtil {
+
+    public static boolean configureGlobalTracer(OxdServerConfiguration configuration, String componentName) {
+        String tracerName = configuration.getTracer();
+        Tracer tracer = null;
+        if (!configuration.getEnableTracing() || Strings.isNullOrEmpty(tracerName)) {
+            tracer = NoopTracerFactory.create();
+        } else if ("jaeger".equals(tracerName)) {
+            Configuration.SamplerConfiguration samplerConfig = new Configuration.SamplerConfiguration()
+                    .withType(ConstSampler.TYPE)
+                    .withParam(1);
+            Configuration.SenderConfiguration senderConfig = new Configuration.SenderConfiguration()
+                    .withAgentHost(configuration.getTracerHost())
+                    .withAgentPort(configuration.getTracerPort());
+            Configuration.ReporterConfiguration reporterConfig = new Configuration.ReporterConfiguration()
+                    .withLogSpans(true)
+                    .withFlushInterval(1000)
+                    .withMaxQueueSize(10000)
+                    .withSender(senderConfig);
+            tracer = new Configuration(componentName).withSampler(samplerConfig).withReporter(reporterConfig).getTracer();
+        } else if ("zipkin".equals(tracerName)) {
+            OkHttpSender sender = OkHttpSender.create(
+                    "http://" +
+                            configuration.getTracerHost() + ":" +
+                            configuration.getTracerPort() + "/api/v1/spans");
+            Reporter<Span> reporter = AsyncReporter.builder(sender).build();
+            tracer = BraveTracer.create(Tracing.newBuilder()
+                    .localServiceName(componentName)
+                    .spanReporter(reporter)
+                    .build());
+        } else {
+            tracer = NoopTracerFactory.create();
+        }
+        GlobalTracer.register(tracer);
+        return true;
+    }
+
+    private static Tracer getGlobalTracer() {
+        return GlobalTracer.get();
+    }
+
+    public static Scope buildSpan(String spanName, boolean startActive) {
+        return getGlobalTracer().buildSpan(spanName).startActive(startActive);
+    }
+
+    public static io.opentracing.Span buildChildSpan(io.opentracing.Span parentSpan, String childSpanName) {
+        return getGlobalTracer().buildSpan(childSpanName).asChildOf(parentSpan).start();
+    }
+
+    public static io.opentracing.Span getActiveSpan() {
+        return getGlobalTracer().activeSpan();
+    }
+
+    public static void setTag(String tagName, String tagValue) {
+        if(getGlobalTracer().activeSpan() != null) {
+            getGlobalTracer().activeSpan().setTag(tagName, tagValue);
+        }
+    }
+
+    public static void log(String message) {
+        if(getActiveSpan() != null) {
+            getActiveSpan().log(message);
+        }
+    }
+
+    public static void errorLog(Exception e) {
```

Comment:
Shouldn't we check here whether tracing is enabled, if no then directly return ?

Reply:
Here we are adding tracer from `NoopTracerFactory` if tracing is not enabled.
https://github.com/GluuFederation/oxd/blob/issue%23182/oxd-server/src/main/java/org/gluu/oxd/server/TracingUtil.java#L34

Now `getActiveSpan()` method will return null if tracing is not enabled and we are checking if activeSpan is not null.

https://github.com/GluuFederation/oxd/blob/issue%23182/oxd-server/src/main/java/org/gluu/oxd/server/TracingUtil.java#L108

We have used this approach as advised here: https://stackoverflow.com/questions/54134885/disable-opentracing-globally

Reply:
I see, agreed.